Lib/functools.py

```diff
@@ -412,11 +409,10 @@ def __setstate__(self, state):
            (namespace is not None and not isinstance(namespace, dict))):
             raise TypeError("invalid partial state")
 
-        if args and args[-1] is Placeholder:
-            raise TypeError("trailing Placeholders are not allowed")
         phcount, merger = _partial_prepare_merger(args)
 
-        args = tuple(args) # just in case it's a subclass
+        if type(args) is not tuple:
+            args = tuple(args)       # just in case it's a subclass
```

Comment:
```suggestion
        args = tuple(args)       # just in case it's a subclass
```
This is faster and more readable. For an exact tuple no copy is made.

Reply:
Thanks, fixed it.